pandas/tests/io/excel/test_readers.py

```diff
@@ -1229,3 +1227,10 @@ def test_read_datetime_multiindex(self, engine, read_ext):
         expected = DataFrame([], columns=expected_column_index)
 
         tm.assert_frame_equal(expected, actual)
+
+    def test_engine_invalid_option(self, read_ext):
+        # read_ext includes the '.' hence the weird formatting
+        engine = pd.get_option(f"io.excel{read_ext}.reader")
```

Comment:
can you instead use the context manager to test options. this is likely causing side effects.

Reply:
@jreback can you point me to some documentation on how to use this? I am currently following the approach used here. 
https://github.com/pandas-dev/pandas/pull/31569/files#diff-5974844fa02c9b81f3477198023bd02503bf82da1e3d15df6dfd62cdca48666bR242

Reply:
use pd.option_context

Reply:
https://github.com/pandas-dev/pandas/issues/38813 that is not a good example to follow